### ISSUE TITLE
recover training: key error

### ISSUE THREAD

**Issue by shuningjin** [allenai/vampire#53]
When recovering prior vampire pertaining or classifier training, I encountered key errors:

Key [random_seed, pytorch_seed, numpy_seed] found in training configuration but not in the serialization directory we're recovering from.

----
Version: PyTorch 1.2.0, AllenNLP 0.8.4

Pretraining command:
```
export DATA_DIR="$(pwd)/examples/ag"
export VOCAB_SIZE=30000
python -m scripts.train \
            --config training_config/vampire.jsonnet \
            --serialization-dir model_logs/ag \
            --environment VAMPIRE \
            --device 0 \
            --seed 1 \
            --recover	 
```
Pretraining error:

```
2020-01-15 12:45:52,302 - INFO - allennlp.training.util - Recovering from prior training at model_logs/ag_1.
2020-01-15 12:45:52,324 - ERROR - allennlp.training.util - Key 'random_seed' found in training configuration but not in the serialization directory we're recovering from.
2020-01-15 12:45:52,324 - ERROR - allennlp.training.util - Key 'pytorch_seed' found in training configuration but not in the serialization directory we're recovering from.
2020-01-15 12:45:52,324 - ERROR - allennlp.training.util - Key 'numpy_seed' found in training configuration but not in the serialization directory we're recovering from.
```

Classification command:

```
export VAMPIRE_DIR="$(pwd)/model_logs/vampire"
export VAMPIRE_DIM=81
export THROTTLE=200
export EVALUATE_ON_TEST=1
python -m scripts.train \
            --config training_config/classifier.jsonnet \
            --serialization-dir model_logs/clf \
            --environment CLASSIFIER \
            --device 0 \
            --seed 1 \
	    --recover	
````

Classification error:

```
2020-01-15 12:52:46,094 - INFO - allennlp.training.util - Recovering from prior training at model_logs/clf_1.
2020-01-15 12:52:46,112 - ERROR - allennlp.training.util - Key 'pytorch_seed' found in training configuration but not in the serialization directory we're recovering from.
2020-01-15 12:52:46,112 - ERROR - allennlp.training.util - Key 'numpy_seed' found in training configuration but not in the serialization directory we're recovering from.
2020-01-15 12:52:46,112 - ERROR - allennlp.training.util - Key 'random_seed' found in training configuration but not in the serialization directory we're recovering from.
```

**Comment by kernelmachine** [comment]
Thanks for reporting this bug! It looks like the issue has to do with AllenNLP, as it doesn't seem to save the seed in the serialization directory. 

As a quick fix, you can manually append the desired random/pytorch/numpy seed in the `config.json` file of the serialization directory:
```
{
"pytorch_seed": "1",
"random_seed": "1",
"numpy_seed": "1"
...
}
```

Then comment out [this](https://github.com/allenai/vampire/blob/master/scripts/train.py#L61) line, and run
 
```
python -m scripts.train \
            --config training_config/vampire.jsonnet \
            --serialization-dir model_logs/ag_1 \
            --environment VAMPIRE \
            --device 0 \
            --seed 1 \
            --recover	
```
